### DISCUSSION TITLE
How to add a custom title bar in electron in quasar when using typescript.

### DISCUSSION THREAD

**Discussion by Nethrenial** [quasarframework/quasar#9883] [CLI - Electron mode]
Please, can any of you help me, after setting my quasar project with electron and typescript, I tried to add a custom title bar with help of an official example in quasar documentation, in it, we inject a new custom myWindowApi to window object in renderer thread via remote and ContextBridge.While this works in development mode and title bar functions as intended(even then Vetur and ESLint throw errors). But when I try to build it compilation fails because the property 'myWindowApi' doesn't exist on the window. I can't get to typescript to recognize my custom injected property in the window object. I'm so frustrated because I followed the official docs correctly
![Screenshot (180)](https://user-images.githubusercontent.com/67809946/124006253-a4cd4e80-d9f7-11eb-8db8-186f27f7b555.png)
.
![Screenshot (179)](https://user-images.githubusercontent.com/67809946/124006047-69cb1b00-d9f7-11eb-8098-b0ee544156f6.png)

**Comment by ontwikkelfabriek** [comment]
@Nethrenial 

>  I tried to add a custom title bar with help of an official example in quasar documentation

What example?

**Reply by Nethrenial** [reply]
https://quasar.dev/quasar-cli/developing-electron-apps/frameless-electron-window 
This but this uses Javascript so it doesn't give any warning when I access my custom api in the window object,is there any way I can do it with typescript.

**Reply by ontwikkelfabriek** [reply]
@Nethrenial 

I am not a typescript expert. but can't you define an interface for the `window` type with a `myWindowsAPI` property....

**Reply by tgarif** [reply]
@Nethrenial 

Have you found a way to achieve it using typescript? I'm also struggling with the warning...

**Comment by hawkeye64** [comment]
Did you do the preload script properly? https://quasar.dev/quasar-cli/developing-electron-apps/frameless-electron-window#the-preload-script
